src/libs/rover/rover.cpp

```diff
@@ -26,6 +26,12 @@ Rover::Rover()
 {
   // Ensure that we always start from a default state
   rover::settings.reset();
+
+  // Metadata
+  rover::settings["state/time"] = -1.0f;
+  rover::settings["state/cycle"] = -1;
```

Comment:
I'm not sure how I feel about these default values. It probably makes more sense to not include these by default, and only add them if the dataset includes them.

You should craft a test case that does not have cycle or time and see what happens.